src/pyaro/timeseries/Filter.py

```diff
@@ -1184,3 +1184,162 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+                )
+        return self._topography
+
+    def init_kwargs(self):
+        return {
+            "topo_file": self._topo_file,
+            "topo_var": self._topo_var,
+            "radius": self._radius,
+            "lower": self._lower,
+            "upper": self._upper,
+        }
+
+    def name(self):
+        return "valleyfloorrelativealtitudefilter"
+
+    def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+        filtered_stations = {}
+        for k, v in stations.items():
+            lat = v.latitude
+            lon = v.longitude
+            alt = v.altitude
+
+            ralt = self._calculate_relative_altitude(
+                lat, lon, radius=self._radius, altitude=alt
+            )
+
+            keep = True
+            if self._lower is not None:
+                if self._lower > ralt:
+                    keep = False
+            if self._upper is not None:
+                if self._upper < ralt:
+                    keep = False
+            if keep:
+                filtered_stations[k] = v
+
+        return filtered_stations
+
+    def _calculate_relative_altitude(
+        self, lat: float, lon: float, *, radius: float, altitude: float
+    ):
+        # At most one degree of latitude (at equator) is roughly 111km.
+        # Subsetting to based on this value with safety margin makes the
+        # distance calculation MUCH more efficient.
+        s = 0.1 + (radius / 1000) / 100
+        topo = self.topography.sel(
+            lon=slice(lon - s, lon + s), lat=slice(lat - s, lat + s)
+        )
+
+        distances = self._haversine(topo["lon"], topo["lat"], lon, lat)
+        within_radius = distances <= radius
+
+        values_within_radius = topo[self._topo_var].where(within_radius, drop=True)
+
+        min_value = float(values_within_radius.min())
+        return altitude - max([min_value, 0])
+
+    def _find_lat_lon_variables(self, topo_xr):
+        """
+        Find and load DataArrays from topo which represent the latitude and longitude
+        dimensions in the topography data.
+
+        These are assigned to self._lat, self._lon, respectively for later use.
+
+        :param topo_xr xr.Dataset of topography
+        :return lat, lon DataArrays
+        """
+        lat = None
+        lon = None
+        for var_name in topo_xr:
+            unit_str = topo_xr[var_name].attrs.get("units", None)
+            if unit_str in self.UNITS_LAT:
+                lat = topo_xr[var_name]
+                continue
+            if unit_str in self.UNITS_LON:
+                lon = topo_xr[var_name]
+                continue
+
+        if any(x is None for x in [lat, lon]):
+            raise ValueError(
+                f"Required variable names for lat, lon dimensions could not be found in file '{self._topo_file}"
+            )
+        return lat, lon
+
+    def _haversine(self, lon1, lat1, lon2, lat2):
```

Comment:
haversine does not depend on self. This functions belongs into a helper-file (e.g. math_funcs), and doesn't need to be private.